sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -895,4 +897,14 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       .orderBy(sum('j))
     checkAnswer(query, Row(1, 2))
   }
+
+  test("SPARK-10316: respect non-deterministic expressions in PhysicalOperation") {
```

Comment:
@cloud-fan 
This test case is no longer valid since "push `Filter` through `project` never happen" in this case. (latest master code)
This mean even if I do below change, the case still can pass
```
-      case Project(fields, child) if fields.forall(_.deterministic) =>
+      case Project(fields, child) =>
...
-      case Filter(condition, child) if condition.deterministic =>
+      case Filter(condition, child) =>
```
Should I submit a new test case for it?

Reply:
sure, thanks!